Add a showcase page for high-quality, published games

Calinou opened the pull request:
This adds a revamped showcase page designed to host high-quality, published projects. (It's not limited to games, even though most projects featured here will naturally be games.)

It now uses richer formatting and allows for better showcasing projects with a screenshot and optional YouTube video.

The content is managed via static pages (committed to this repository) that make use of a predefined layout.

## TODO

While the frontend side is mostly done, there's quite a lot of stuff to do on the content front:

- [x] Reference the showcase prominently on the homepage or elsewhere. Since the primary navigation bar is quite full already, I added a link to the showcase in the secondary navigation instead (not shown on the screenshots below).
- [x] [Contact developers with existing high-quality published games to be included in this showcase.
  We should aim to have at least 15-20 projects listed when the showcase is first published.
- [x] Publish a form that can be used by developers to submit their projects.
  - This form will also allow developers to leave their email address to be contacted for an optional interview; in case they want to talk about their experience with Godot in a blog post on the website. This blog post will then be quoted and referenced on the showcase's project page.

## Preview

### Showcase list

![Screenshot_2020-11-06 Godot Engine - Showcase(1)](https://user-images.githubusercontent.com/180032/98377781-84eeb200-2045-11eb-9a50-3cf0cacca4d3.png)

### Showcase item

![Screenshot_2020-11-06 Godot Engine - Showcase Example Game](https://user-images.githubusercontent.com/180032/98379491-a6509d80-2047-11eb-9e74-402aef2b9a46.png)

Calinou committed:
Add a showcase page for high-quality, published games

nikhilCad commented:
I think Godot's showcase Youtube videos should also be at the top. Look how Gamemaker links to their Youtube video in a fancy way https://www.yoyogames.com/showcase

YuriSizov commented on themes/godotengine/pages/showcase/example-game.htm:
```diff
@@ -0,0 +1,58 @@
+{% endput %}
+
+{% put image %} https://cdn.cloudflare.steamstatic.com/steam/apps/404790/ss_f536b553fe0ec777c6ff89e968ac17b9d79341aa.jpg {% endput image %}
+{% put youtube_embed %} https://www.youtube-nocookie.com/embed/UEDEIksGEjQ {% endput %}
```
Since it's a `youtube_embed` param already, maybe just require a video id and put it into the nocookie host in the template automatically?

YuriSizov commented on themes/godotengine/pages/showcase/example-game.htm:
```diff
@@ -0,0 +1,58 @@
+
+{% put image %} https://cdn.cloudflare.steamstatic.com/steam/apps/404790/ss_f536b553fe0ec777c6ff89e968ac17b9d79341aa.jpg {% endput image %}
+{% put youtube_embed %} https://www.youtube-nocookie.com/embed/UEDEIksGEjQ {% endput %}
+{% put link %} https://store.steampowered.com/app/404790/ {% endput %}
```
Probably makes sense to support multiple links? Not sure if `put`/`placeholder` would allow us that. Maybe we can bake in special `link_steam`, `link_gog`, `link_egs`, etc and allow one `link_custom` for something unsupported, like a personal page with the humble widget.

Calinou commented:
I added the showreel videos at the top of the showcase page. This time, the iframes will automatically use a 16:9 aspect ratio.

I also added the ability to have multiple store links, and made the YouTube embed require only specifying the embed code instead of the full embed URL.

See the first post for updated screenshots.

gdesecrate commented:
Why not also make a showcase for tutorials and courses(both free and paid) as well?

YuriSizov commented:
@gdesecrate Some collection of tutorials would be cool (and resolve the issue of the tutorial channel on Discord being useless), but this is a separate subject and you should open a new issue for it.

Calinou commented:
@gdesecrate We already have a documentation page listing third-party tutorials and resources: https://docs.godotengine.org/en/latest/community/tutorials.html

gdesecrate commented:
@Calinou I mean the featured showcase with big thumbnails for videos/courses which easy to reach from main site.
For example, I never knew that the page you linked existed.
It seems to reach your page user needs to press [learn] on site, scroll down a lot of menu options and press particular link. (Correct me if I wrong)
I believe tutorials deserve an easy-to-reach showcase as well.

Calinou commented:
@gdesecrate I opened a PR to improve discoverability of the tutorials page: https://github.com/godotengine/godot-website/pull/202

Calinou committed:
Add the initial projects for the showcase

YuriSizov commented on themes/godotengine/pages/showcase/city-game-studio.htm:
```diff
@@ -0,0 +1,73 @@
+{##}
+{% put title %} City Game Studio {% endput %}
+{% put author %} Binogure Studio {% endput %}
+{% put author_url %} https://hivetime.twolofbees.com/ {% endput %}
```
This is probably a wrong URL, right?

Calinou replied:
Good catch, fixed :slightly_smiling_face:

YuriSizov commented:
@Calinou Any chance to see how the list looks now with some projects added? Also, if we are featuring both games and apps, does it make sense to split them maybe?

Calinou commented:
> Any chance to see how the list looks now with some projects added?

<details>

![image](https://user-images.githubusercontent.com/180032/99561590-68e4fc00-29c7-11eb-9cf9-575d363e2fd5.png)
</details>

> Also, if we are featuring both games and apps, does it make sense to split them maybe?

There won't be that many applications, so I'm not sure if it's worth splitting them.

YuriSizov commented:
Neat!

Another thought: does it make sense to show which version of the engine was used?

Calinou commented:
> Another thought: does it make sense to show which version of the engine was used?

I discussed it with Akien and decided that it wasn't particularly useful to mention.

akien-mga commented on themes/godotengine/assets/icons/ios.svg:
```diff
@@ -0,0 +1 @@
+<svg aria-hidden="true" focusable="false" data-prefix="fab" data-icon="app-store-ios" class="svg-inline--fa fa-app-store-ios fa-w-14" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM127 384.5c-5.5 9.6-17.8 12.8-27.3 7.3-9.6-5.5-12.8-17.8-7.3-27 ...
```
That seems to be the logo of the App Store? It's not very recognizable to me as a non iOS / macOS user, it might be better to use the iOS logo? https://en.wikipedia.org/wiki/IOS#/media/File:IOS_wordmark_(2017).svg

akien-mga commented on themes/godotengine/assets/icons/linux.svg:
```diff
@@ -0,0 +1 @@
+<svg aria-hidden="true" focusable="false" data-prefix="fab" data-icon="linux" class="svg-inline--fa fa-linux fa-w-14" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path fill="currentColor" d="M220.8 123.3c1 .5 1.8 1.7 3 1.7 1.1 0 2.8-.4 2.9-1.5.2-1.4-1.9-2.3-3.2-2.9-1.7-.7-3.9-1-5.5-.1-.4.2-.8.7-.6 1.1.3 1.3 2.3 1.1 3.4 1.7zm-21.9 1.7c1.2 0 2-1.2 3-1.7 1.1-.6 3.1-.4 3.5-1.6 ...
```
We could alternatively use Stycil Tux which IMO is worth promoting over Linus' duck-penguin :) https://www.twolofbees.com/artwork.php?iid=870

akien-mga commented:
Looks good to me. We might want to refine the design a bit to make things more fancy looking, but it's a good base.

So we go for a static content approach for all showcase entries over integrating with the CMS and database I guess? The latter might be easier to add/amend entries, but requires some extra work on making a plugin similar to the old portfolio one.

YuriSizov commented:
I assume the CMS route would prevent easy contributions, would it not?

Calinou commented:
> I assume the CMS route would prevent easy contributions, would it not?

It doesn't really make a difference. We'd be adding the showcase information ourselves every time, as pull requests won't be used for the showcase.

Still, creating a custom content type in October doesn't look that trivial, so I'd prefer to keep using static files for now.

Calinou commented:
New icons for Linux and iOS added:

![image](https://user-images.githubusercontent.com/180032/100110127-e481f580-2e6c-11eb-87eb-506b41306a39.png)

Thanks @Cheeseness for providing the Linux icon :slightly_smiling_face: 

~~I still have to add icons for consoles.~~

**Edit:** Added icons for consoles and increased spacing between icons:

![image](https://user-images.githubusercontent.com/180032/100126199-f28c4200-2e7d-11eb-8f87-44dec3a173ef.png)

akien-mga reviewed (approved):
Awesome work both on the frontend and collecting all these entries! Let's go live :)

akien-mga commented:
Thanks!
